src/components/PostsList.tsx

```diff
@@ -1,86 +1,70 @@
 import React from 'react';
+import classNames from 'classnames';
+import { Post } from '../types/Post';
 
-export const PostsList: React.FC = () => (
-  <div data-cy="PostsList">
-    <p className="title">Posts:</p>
+type Props = {
+  isPostsList: false | Post[] | null;
+  posts: Post[];
+  setActivePost: (v: Post | null) => void;
+  activePost: Post | null;
+};
 
-    <table className="table is-fullwidth is-striped is-hoverable is-narrow">
-      <thead>
-        <tr className="has-background-link-light">
-          <th>#</th>
-          <th>Title</th>
-          {/* eslint-disable-next-line jsx-a11y/control-has-associated-label */}
-          <th> </th>
-        </tr>
-      </thead>
+export const PostsList: React.FC<Props> = ({
+  isPostsList,
+  posts,
+  setActivePost,
+  activePost,
+}) => {
+  if (!isPostsList || isPostsList.length === 0) {
```

Comment:
The condition `!isPostsList || isPostsList.length === 0` is incorrect due to the improper typing of `isPostsList`. Once `isPostsList` is correctly typed as a boolean, this condition should be revised to `!isPostsList`.